plutus-metatheory/src/Algorithmic/CK.lagda.md

```diff
@@ -70,33 +71,33 @@ discharge : ∀{A : ∅ ⊢Nf⋆ *}{t : ∅ ⊢ A} → Value t → ∅ ⊢ A
 discharge {t = t} _ = t
 
 step : ∀{A} → State A → State A
-step (s ▻ ƛ L)                    = s ◅ V-ƛ L
-step (s ▻ (L · M))                = (s , -· M) ▻ L
-step (s ▻ Λ L)                    = s ◅ V-Λ L
-step (s ▻ (L ·⋆ A / refl))        = (s , -·⋆ A) ▻ L
-step (s ▻ wrap A B L)             = (s , wrap-) ▻ L
-step (s ▻ unwrap L refl)          = (s , unwrap-) ▻ L
-step (s ▻ con cn)                 = s ◅ V-con cn
-step (s ▻ error A)                = ◆ A
-step (ε ◅ V)                      = □ V
-step ((s , (-· M)) ◅ V)           = ((s , V ·-) ▻ M)
-step ((s , (V-ƛ t ·-)) ◅ V)       = s ▻ (t [ discharge V ])
-step ((s , (-·⋆ A)) ◅ V-Λ t)      = s ▻ (t [ A ]⋆)
-step ((s , wrap-) ◅ V)            = s ◅ (V-wrap V)
-step ((s , unwrap-) ◅ V-wrap V)   = s ▻ deval V
-step (s ▻ (builtin b / refl))     = s ◅ ival b
-step ((s , (V-I⇒ b {am = 0} bt ·-)) ◅ vu) = s ▻ BUILTIN' b (app bt vu)
+step (s ▻ ƛ L)                                = s ◅ V-ƛ L
+step (s ▻ (L · M))                            = (s , -· M) ▻ L
+step (s ▻ Λ L)                                = s ◅ V-Λ L
+step (s ▻ (L ·⋆ A / refl))                    = (s , -·⋆ A) ▻ L
```

Comment:
pattern synonym?

Reply:
I'm not convinced we should make every constructor with a `refl` a pattern synonym. Unlike constructors, pattern synonyms have poor overloading of names, so they are best used local to a module. We reuse a lot of syntax, and if we took the route of doing a pattern synonym to avoid every `refl`, we would need unique global names for every pattern synonym. This means introducing a lot of new names just for the sake of avoiding the ocasional `refl`.